pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -153,7 +153,7 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 	) -> Result<ExecutedForeignCollectInvest<T::Balance>, DispatchError> {
 		// No need to transition or update state as collection of tranche tokens is
 		// independent of the current `InvestState`
-		let CollectedAmount::<T::Balance> {
+		let CollectedInvestment::<T::Balance> {
```

Comment:
The left todo, regarding `pool_currency` and `return_currency`. 

* How would we know what the inital `return_curreny` is? I.e. the inital investment currency? 
  * what if the conversion rate changes in between?

Reply:
We decided on adding the `return_currency` to the `CollectInvest` message which can be handled entirely synchronously!